tests/integration/test_opensearch.py

```diff
@@ -491,6 +491,31 @@ def test_route_through_edge(self):
             lambda: not cluster.is_up(), timeout=240
         ), "gave up waiting for cluster to shut down"
 
+    def test_custom_endpoint(self, opensearch_client, opensearch_wait_for_cluster):
+        domain_name = f"opensearch-domain-{short_uid()}"
+        custom_endpoint = "http://localhost:4566/my-custom-endpoint"
+        domain_endpoint_options = {
+            "CustomEndpoint": custom_endpoint,
+            "CustomEndpointEnabled": True,
```

Comment:
If you disable the Custom Endpoint here (by setting `"CustomEndpointEnabled": False`) the routing will break (since `build_cluster_endpoint` will _not_ return `custom_endpoint.endpoint` but the URL based on the strategy).
This is non-obvious and:
- Needs a test
- Should be documented in the `register_cluster ` function (that it's tightly coupled to the `build_cluster_endpoint` function.

Reply:
:+1: